Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_wfsi.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <string>
+
+#include <mbedtls/aes.h>
+
+#include "Common/CommonTypes.h"
+#include "Core/IPC_HLE/WII_IPC_HLE.h"
+#include "Core/IPC_HLE/WII_IPC_HLE_Device.h"
+
+class ARCUnpacker
```

Comment:
Probably. Just like TMD, ticket and content handling code should be factored out and put in one place...